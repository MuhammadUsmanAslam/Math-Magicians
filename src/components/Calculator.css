.calculator {
  width: 45%;
}

.calculator__input {
  font-size: 32px;
  outline: none;
  width: 100%;
  padding: 0.5em;
  text-align: right;
  background-color: #858694;
  border: 1px solid #b9b8b8;
}

.calculator__row {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr;
}

.calculator__row:last-child {
  grid-template-columns: 2fr 1fr 1fr;
}

.calculator__row > button {
  padding: 1em 0;
  font-weight: 500;
  font-size: 20px;
  background-color: #e0e0e0;
  border: 1px solid #b9b8b8;
  cursor: pointer;
}

.calculator__row > button:last-child {
  background-color: #f5913e;
}
