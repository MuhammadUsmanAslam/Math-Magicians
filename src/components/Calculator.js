import './Calculator.css';
import CalculatorRow from './CalculatorRow.js';

function Calculator() {
    const RowData = [["AC", "+/-", "%", "รท"], ["7", "8", "9", "x"], ["4", "5", "6", "-"], ["1", "2", "3", "+"], ["0", ".", "="]];
    return (
        <div className="calculator">
            <input className='calculator__input' type="text" defaultValue={0} />
            {RowData.map((RowDataItem, index) => {
                return <CalculatorRow key={index} data={RowDataItem} />
            })}
        </div >
    );
}

export default Calculator;
