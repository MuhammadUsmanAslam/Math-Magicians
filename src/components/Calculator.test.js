import renderer from "react-test-renderer";
import Calculator from "./Calculator";

test("render calculator component", () => {
	const tree = renderer.create(<Calculator></Calculator>).toJSON();

	expect(tree).toMatchSnapshot();
});
