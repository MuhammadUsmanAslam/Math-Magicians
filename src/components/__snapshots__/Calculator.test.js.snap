// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render calculator component 1`] = `
<div
  className="calculator"
>
  <div
    className="calculator__input"
  />
  <div
    className="calculator__row"
  >
    <button
      name="AC"
      onClick={[Function]}
    >
      AC
    </button>
    <button
      name="+/-"
      onClick={[Function]}
    >
      +/-
    </button>
    <button
      name="%"
      onClick={[Function]}
    >
      %
    </button>
    <button
      name="รท"
      onClick={[Function]}
    >
      รท
    </button>
  </div>
  <div
    className="calculator__row"
  >
    <button
      name="7"
      onClick={[Function]}
    >
      7
    </button>
    <button
      name="8"
      onClick={[Function]}
    >
      8
    </button>
    <button
      name="9"
      onClick={[Function]}
    >
      9
    </button>
    <button
      name="x"
      onClick={[Function]}
    >
      x
    </button>
  </div>
  <div
    className="calculator__row"
  >
    <button
      name="4"
      onClick={[Function]}
    >
      4
    </button>
    <button
      name="5"
      onClick={[Function]}
    >
      5
    </button>
    <button
      name="6"
      onClick={[Function]}
    >
      6
    </button>
    <button
      name="-"
      onClick={[Function]}
    >
      -
    </button>
  </div>
  <div
    className="calculator__row"
  >
    <button
      name="1"
      onClick={[Function]}
    >
      1
    </button>
    <button
      name="2"
      onClick={[Function]}
    >
      2
    </button>
    <button
      name="3"
      onClick={[Function]}
    >
      3
    </button>
    <button
      name="+"
      onClick={[Function]}
    >
      +
    </button>
  </div>
  <div
    className="calculator__row"
  >
    <button
      name="0"
      onClick={[Function]}
    >
      0
    </button>
    <button
      name="."
      onClick={[Function]}
    >
      .
    </button>
    <button
      name="="
      onClick={[Function]}
    >
      =
    </button>
  </div>
</div>
`;
